src/harness/testcases/WINNF_FT_S_MCP_testcase.py

```diff
@@ -0,0 +1,328 @@
+			'cert_file': 'certs/server_b.cert',
+			'key_file': 'certs/server_b.key'
+		        }],
+        'sas_uut': {'name': 'SAS-UUT',
+                        'url': ['localhost', '9000'],
+                        'cert_file': 'certs/server.cert',
+                        'key_file': 'certs/server.key'}, 
+        'K':3, 
+        'headroom':{'MgPpa':2,'MgGwpz':2,'MgCochannel':2,'MgBlocking':2,'MgOobe':2,'MgEsc':2},
+        'DPAactiveInfo':[],
+        'DPAdeactiveInfo':[]
+       }
+    
+    writeConfig(filename, config)
+
+
+  @configurable_testcase(generate_MCP_1_default_config) 
+  def test_WINNF_FT_S_MCP_1(self,config_filename):
+    """SAS manages a mix of GAA and PAL Grants in 3550 MHz
+       to 3700 MHz to protect configurable IAP-protected entities and DPAs
+    """
+    config = loadConfig(config_filename)
+    pre_iap_headroom = config ['headroom']
+
+    
+    #Step 1 : Load DPAs
+    self._sas_admin.TriggerLoadDpas()
+
+    #STEP 2 : ESC informs SAS about inactive DPA
+    self._sas_admin.TriggerBulkDpaActivation({'activate':False})
+
+    #Step 3 : creates multiple SAS TH, and Load predefined FAD,CBSD 
+    self.createSASTH(config)
+    self.configureFadForEachSASTH(0)
+    
+    #fad objects get the fad dumps from the remote SAS configured with 'url'
+    fad_object_th = []
+    for N in len(config['sas_hareness']):
+       cert = os.path.join('certs',config['sas_hareness'][N]['cert_file'])
+       key  = os.path.join('certs',config['sas_hareness'][N]['key_file'])
+       fad_object_th.append(FullActivityDump(cert,key,config['sas_hareness'][N]['url']))
+       fad_records.append(fad_object_th[N])
+
+    #for UUT creating fad object
+    cert = os.path.join('certs',config['sas_uut']['cert_file'])
+    key  = os.path.join('certs',config['sas_uut']['key_file'])
+    fad_uut_object = FullActivityDump(cert,key,config['sas_uut']['url'])
+    fad_records.append(fad_uut_object)
+
+    # informing SAS UUT about SAS TH 
+    for N in len(config['sas_hareness']):
+       self._sas_admin.InjectPeerSas({'certificateHash': config[N]['certificateHash'],
+                                     'url': 'https://' + config[N] ['url']+ '/dump'}) 
+    
+
+    #Step 4,5 : Inject IAP protected entities into UUT
+    for k in  config['K']:	
+       for records in config['IAPRecords'][k]:
+          if records['type'] == "FSS":
+             self._sas_admin.InjectFss(records['Record'])
+          if records['type'] == "GWPZ":
+             self._sas_admin.InjectWisp(records['Record'])
+          if records['type'] == "ESC":
+             self._sas_admin.InjectEsc(records['Record'])
+          if records['type'] == "PPA":
+             self._sas_admin.InjectZoneData(records['Record'])
+
+     
+        
+       # Step 6 : Trigger FAD generation with SAS test harness
+       if len(config['sas_hareness']) > 0 :
+          self._sas_admin.TriggerFullActivityDump() # triggers to UUT
+
+          # Step 7 : Pull FAD records from SAS UUT
```

Comment:
Please check the test spec -- each SAS TH has to pull from SAS UUT.

Reply:
Require   changes in SAS Test Harness implementation. We will address this comment as   part of SAS TH development activity